engine-tests/src/test_utils/mocked_external.rs

```diff
@@ -0,0 +1,176 @@
+use near_vm_logic::mocks::mock_external::MockedExternal;
+
+pub(crate) struct MockedExternalWithTrie<'a> {
+    underlying: &'a mut MockedExternal,
+    trie_node_count: std::cell::Cell<u64>,
+}
+
+impl<'a> MockedExternalWithTrie<'a> {
+    pub fn new(ext: &'a mut MockedExternal) -> Self {
+        Self {
+            underlying: ext,
+            trie_node_count: std::cell::Cell::new(0),
+        }
+    }
+
+    fn increment_trie_node_count(&self, amount: u64) {
+        let cell_value = self.trie_node_count.get();
+        self.trie_node_count.set(cell_value + amount);
+    }
+}
+
+impl<'b> near_vm_logic::External for MockedExternalWithTrie<'b> {
+    fn storage_set(&mut self, key: &[u8], value: &[u8]) -> Result<(), near_vm_logic::VMLogicError> {
+        self.increment_trie_node_count(10);
```

Comment:
Should we make named constants out of this values. So it is clear where are they coming from!